### ISSUE TITLE
Refactor find_LCDS and find_LCDS_K

### ISSUE THREAD

**Issue by jtniehof** [spacepy/spacepy#768]
Depends on #749.

find_LCDS and find_LCDS_K have several instances of a pattern:

1. unpack prep_irbem values
2. call find_magequator
3. filter bad output

This can probably be made a helper function, either a nested function for each or a separate one that's shared. (First eyeball glance indicates this section is the same between them).

**Comment by drsteve** [comment]
Agreed, but I think this is more broad than just `find_magequator`. The same pattern is there for other irbem calls too...

As a side note, we should think about deprecating `find_LCDS` as it's basically an implementation of an incorrect method that had previous use in the literature. It was there basically to provide a way to illustrate the differences.

_Short explanation: L* is usually calculated as a function of pitch angle at a given location, but this is used to calculate K which is invariant around the drift shell. Different K at a given location will have a different L*. For an LCDS search you want to keep K invariant so that you're comparing like with like as you do your bisection to find the last valid L* (aka LCDS). The constant-alpha approach in `find_LCDS` means each search has a different K._

**Comment by jtniehof** [comment]
Mostly just identifying low hanging fruit.

At a minimum we should updte that docstring to warn people off. Should we put in specific citations for where the inaccurate one was used? I can see preserving it for reproduction more or less indefinitely.

I can't seem to find my copy of Roederer but I presume that would be reasonable cite for a fuller explanation.

**Comment by drsteve** [comment]
Most specific explanation for LCDS in particular is in Albert et al. (2018; https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2018JA025991). But yeah, Roederer is good for the underlying issues.

**Comment by jamesw732** [comment]
I realized I didn't make #769 close this when it was supposed to. Deprecating `find_LCDS` seems to me like a separate issue, so I'll call this done and close it.
